src/class-post-type.php

```diff
@@ -0,0 +1,85 @@
+			'menu_name'             => __( $plural, $translation_domain ),
+			'name_admin_bar'        => __( $singular, $translation_domain ),
+			'all_items'             => __( 'All ' . $plural_lowercase, $translation_domain ),
+			'add_new_item'          => __( 'Add a ' . $singular_lowercase, $translation_domain ),
+			'edit_item'             => __( 'Edit ' . $singular_lowercase, $translation_domain ),
+			'new_item'              => __( 'New ' . $singular_lowercase, $translation_domain ),
+			'view_item'             => __( 'See ' . $singular_lowercase, $translation_domain ),
+			'view_items'            => __( 'See ' . $plural_lowercase, $translation_domain ),
+			'archives'              => __( $plural_lowercase . ' archives', $translation_domain ),
+			'attributes'            => __( $plural_lowercase . ' attributs', $translation_domain ),
+			'filter_items_list'     => __( 'Filter ' . $plural_lowercase . ' list', $translation_domain ),
+			'items_list'            => __( $plural_lowercase . ' list', $translation_domain ),
+			'items_list_navigation' => __( $plural_lowercase . ' list navigation', $translation_domain ),
+			'search_items'          => __( 'Search ' . $plural_lowercase, $translation_domain ),
+			'not_found'             => __( 'No ' . $singular_lowercase, $translation_domain ),
+			'not_found_in_trash'    => __( 'No ' . $singular_lowercase . ' in trash', $translation_domain ),
+		];
+		// phpcs:enable
+
+		return $labels;
+	}
+
+	/**
+	 * Create a custom post type
+	 *
+	 * @param  string       $singular           Singular name.
+	 * @param  string       $plural             Plural name.
+	 * @param  string|null  $translation_domain Translation domain name.
+	 * @param  array        $args               Custom arguments.
+	 * @param  bool|boolean $public             Is post type public?.
+	 * @param  array        $supports           Supported features.
+	 * @return array                            Custom post type
+	 */
+	public static function create( string $singular, string $plural, string $translation_domain = null, array $args = array(), bool $public = true, array $supports = array() ) {
+		$singular_lowercase = strtolower( $singular );
+
+		$labels = self::generate_labels( $singular, $plural, $translation_domain );
+
+		// phpcs:disable WordPress.WP.I18n
+		$default_args = array(
+			'label'              => __( $singular, $translation_domain ),
+			'labels'             => $labels,
+			'public'             => true,
+			'publicly_queryable' => $public,
+			'capability_type'    => 'post',
+			'supports'           => array(
+				'title',
+				'custom-fields',
+			),
+			'has_archive'        => false,
+			'menu_icon'          => 'dashicons-format-aside',
+		);
+		// phpcs:enable
+
+		$args = array_merge( $default_args, $args );
+
+		$post_type = register_post_type( $singular_lowercase, $args );
+
+		if ( count( $supports ) > 0 ) {
+			add_post_type_support( $singular_lowercase, $supports );
+		}
+
+		return $post_type;
```

Comment:
Do we need to return something ?

Reply:
It can be usefull to check if an error occured or to get the current WP_Post_type object to do things with.